### PR TITLE
blobstore feature. [Finishes #146714671]

### PR DESCRIPTION
**Pull request by siennathesane**
Finishes the blobstore delivery story. Squashed a long commit history for brevity.

Signed-off-by: Mike Lloyd <kevin.michael.lloyd@gmail.com>

//cc @sreenivasmrpivot

### CODE REVIEW

File: ci/tasks/configure-ert.sh [hunk 1]
```diff
@@ -33,6 +30,43 @@ perl -pi -e "s/{{pcf_az_3}}/${pcf_az_3}/g" ${json_file}
...
 
 
+if [[ "${azure_access_key}" != "" ]]; then
+cat ${json_file} | jq \
+  --arg azure_access_key ${azure_access_key} \
+  --arg azure_account_name ${azure_account_name} \
+  --arg azure_buildpacks_container ${azure_buildpacks_container} \
+  --arg azure_droplets_container ${azure_droplets_container} \
+  --arg azure_packages_container ${azure_packages_container} \
+  --arg azure_resources_container ${azure_resources_container} \
```
Comment (krishicks) [inline ci/tasks/configure-ert.sh]: These args should all be wrapped in quotes; parameter expansion (`${}`) would mean anyone using spaces in the variable names would get unexpected args.

File: json_templates/azure/c0-azure-base/ert-template.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-    },
-    "other_availability_zones": [
-      {
         "name": "null"
-      }
-    ],
+      },
+      "other_availability_zones": [
+        {
+          "name": "null"
```
Comment (krishicks) [inline json_templates/azure/c0-azure-base/ert-template.json:9]: I see this was null before; why null?

File: json_templates/azure/c0-azure-base/ert-template.json [hunk 3]
```diff
@@ -65,46 +65,325 @@
         "value": "mysql-proxy-lb.sys.{{pcf_ert_domain}}"
       },
       ".mysql_monitor.recipient_email": {
-        "value" : "c0-coreteam@pivotal.io"
+        "value": "c0-coreteam@pivotal.io"
```
Comment (krishicks) [inline json_templates/azure/c0-azure-base/ert-template.json:68]: This should be a param for the pipeline, not defaulted to c0-coreteam@pivotal.io.

File: json_templates/azure/c0-azure-base/ert-template.json [hunk 4]
```diff
@@ -65,46 +65,325 @@
...
-      "smoke-tests": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []},
-      "notifications-ui": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []},
-      "autoscaling-register-broker": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []},
-      "autoscaling-destroy-broker": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []},
-      "push-pivotal-account": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []},
-      "mysql-rejoin-unsafe": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []},
-      "autoscaling": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []},
-      "bootstrap": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []},
-      "notifications": {"instance_type": {"id": "automatic"},"instances": 1,"elb_names": []}
+    "consul_server": {
```
Comment (krishicks) [inline json_templates/azure/c0-azure-base/ert-template.json:73]: Is this just a formatting change or have the values also changed?

**Review by krishicks — CHANGES_REQUESTED** [review]

File: json_templates/azure/c0-azure-base/ert-template.json (hunk 2 above)
Comment (sreenivasmrpivot) [reply json_templates/azure/c0-azure-base/ert-template.json:9]: @krishicks not sure why it was null. We can create a new bug or story in the tracker to get to this. The scope of this story was just to support the azure blob store for ERT tile.

File: json_templates/azure/c0-azure-base/ert-template.json (hunk 3 above)
Comment (sreenivasmrpivot) [reply json_templates/azure/c0-azure-base/ert-template.json:68]: @krishicks This is a pre-existing code we can create a new bug or story in the tracker to get to this. The scope of this story was just to support the azure blob store for ERT tile.

File: json_templates/azure/c0-azure-base/ert-template.json (hunk 4 above)
Comment (sreenivasmrpivot) [reply json_templates/azure/c0-azure-base/ert-template.json:73]: @krishicks i have verified it using BeyondCompare text compare. It seems to be just formatting which has changed. there is no change in the text though.

File: json_templates/azure/c0-azure-base/ert-template.json (hunk 3 above)
Comment (abbyachau) [reply json_templates/azure/c0-azure-base/ert-template.json:68]: Created a tracker story for this: #148097385.

**Pull request merged**